Support 32-bit operating systems

I'm packaging chezmoi for Alpine Linux, and noticed that chezmoi fails to build on 32-bits arches (x86, armv7, armhf):

```
# github.com/twpayne/go-shell
../pkg/mod/github.com/twpayne/go-shell@v0.1.0/shellcgo.go:30:80: cannot use _Ctype_ulong(buflen) (type _Ctype_ulong) as type _Ctype_uint in assignment
```
This error happens when running `go build`.

Build log: https://gitlab.alpinelinux.org/kdaudt/aports/-/jobs/19113

--- a/shellcgo.go
+++ b/shellcgo.go
@@ -27,7 +27,7 @@ func cgoGetUserShell(name string) (string, bool) {
 			result *C.struct_passwd
 		)
 		//nolint:gocritic,staticcheck
-		rc := C.getpwnam_r(cName, &pwd, (*C.char)(unsafe.Pointer(&buf[0])), C.ulong(buflen), &result)
+		rc := C.getpwnam_r(cName, &pwd, (*C.char)(unsafe.Pointer(&buf[0])), C.size_t(buflen), &result)
 		C.free(unsafe.Pointer(cName))
 
 		switch rc {

